src/aaudio/AudioStreamAAudio.cpp

```diff
@@ -357,6 +357,26 @@ Result AudioStreamAAudio::open() {
     return result;
 }
 
+Result AudioStreamAAudio::release() {
+    if (getSdkVersion() < __ANDROID_API_R__) {
+        return Result::ErrorUnimplemented;
+    }
+
+    std::lock_guard<std::mutex> lock(mLock);
+    AAudioStream *stream = mAAudioStream.load();
+    if (stream != nullptr) {
+        if (OboeGlobals::areWorkaroundsEnabled()) {
+            // Make sure we are really stopped. Do it under mLock
+            // so another thread cannot call requestStart() right before the close.
+            requestStop_l(stream);
+            sleepBeforeClose();
```

Comment:
We should not have to sleep here because we are not deleting the stream objects. The autodoc for release() says the callbacks may still be active.